spookyotp/otp.py

```diff
@@ -70,47 +70,59 @@ def _get_algorithm(algorithm_name):
             raise ValueError("Not a valid algorithm: '{}'"
                              .format(algorithm_name))
 
-    def get_qr_code(self):
+    def get_qr_code(self, uri_params=None):
         """
         Return a QR Code (as generated by the qrcode package)
         that can be used to load the parameters onto a phone etc.
         """
-        uri = self.get_uri()
+        uri = self.get_uri(uri_params)
         img = qrcode.make(uri)
         return img
 
-    def save_qr_code(self, filename):
+    def save_qr_code(self, filename, uri_params=None):
         """
         Save a QR Code to the supplied filename. Type is inferred
         from the filename.
         """
-        qr_code = self.get_qr_code()
+        qr_code = self.get_qr_code(uri_params)
         qr_code.save(filename)
 
     @classmethod
-    def _get_uri(cls, secret, issuer, account,
-                 n_digits, algorithm, **other_params):
+    def _get_uri(cls, secret, issuer, account=None,
```

Comment:
What might make more sense is for this to be `_get_uri(cls, **kwargs)`, and then the logic would be:

1. Combine the account and the issuer, if an account was specified
2. Start forming the uri, including the protocol and the account/issuer
3. Append the secret
4. Append `'{0}={1}'.format(key, quote(value))` for everything else in kwargs

And then the public `get_uri` method would handle figuring out which parameters to pass into `_get_uri`.

Something roughly like
```
def get_uri(self, uri_params=None):
    params = {secret: self._secret,
                     issuer: self._issuer}
    for (param, attr) in self._all_parameters:
        if uri_params is None or param in uri_params:
            params[param] = getattr(self, attr)
    return self._get_uri(**params)
```

Does that sound reasonable? That simplifies `_get_uri` by letting it treat most optional parameters equally. The public methods handle translating what the user wants into arguments to `_get_uri`.